File: 800/2040A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n, k;
        cin >> n >> k;

        vector<int> arr(n);
        for (int i = 0; i < n; i++)
            cin >> arr[i];

        bool flag = false;
        int index = -1;

        for (int i = 0; i < n; i++) 
        {
            bool check = true;
            for (int j = 0; j < n; j++) 
            {
                if (i == j) 
                    continue;

                if (abs(arr[i] - arr[j]) % k == 0) 
                {
                    check = false;
                    break;
                }
            }

            if (check) 
            {
                flag = true;
                index = i;
                break;
            }
        }

        if (flag)
            cout << "YES\n" << (index + 1) << "\n";
        else
            cout << "NO\n";
    }

    return 0;
}

File: 800/2040A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int k = sc.nextInt();
            
            int arr[] = new int[n];
            for(int i=0;i<n;i++)
                arr[i] = sc.nextInt();
            
            Boolean flag = false;
            int index = -1;
            
            for(int i=0;i<n;i++)
            {
                Boolean check = true;
                for(int j=0;j<n;j++)
                {
                    if(i == j)
                        continue;
                    
                    if(Math.abs(arr[i]-arr[j]) % k == 0)
                        check = false;
                    if(check == false)
                        break;
                }
                if(check == true)
                {
                    flag = true;
                    index = i;
                    break;
                }
            }
            
            if(flag)
                System.out.println("YES" + "\n" + (index+1));
            else
                System.out.println("NO");
        }
    }
}